src/pepr/operator/controllers/istio/injection.ts

```diff
@@ -29,40 +30,60 @@ export async function enableInjection(pkg: UDSPackage) {
   const annotations = sourceNS.metadata?.annotations || {};
   const pkgKey = `uds.dev/pkg-${pkg.metadata.name}`;
+  const hasMigrationAnnotation = annotations[nativeIstioAnnotation] === "true";
 
   // Mark the original namespace injection setting for if all packages are removed
   if (!annotations[injectionAnnotation]) {
     annotations[injectionAnnotation] = originalInjectionLabel || "non-existent";
   }
 
-  // Ensure the namespace is configured
-  if (!annotations[pkgKey] || originalInjectionLabel !== "enabled") {
-    // Ensure Istio injection is enabled
+  let shouldRestartPods = false;
+
+  // Ensure native Istio migration annotation is present
+  if (!hasMigrationAnnotation) {
+    annotations[nativeIstioAnnotation] = "true";
+    log.info(`Adding native Istio migration annotation to namespace ${pkg.metadata.namespace}.`);
+    shouldRestartPods = true; // Pods need restarting for migration
+  }
+
+  // Ensure Istio injection is enabled
+  if (originalInjectionLabel !== "enabled") {
     labels[injectionLabel] = "enabled";
+    log.info(`Enabling Istio injection for namespace ${pkg.metadata.namespace}.`);
+    shouldRestartPods = true; // Pods need restarting due to label change
+  }
 
-    // Add the package annotation
+  // Ensure package-specific annotation is updated
+  if (annotations[pkgKey] !== "true") {
     annotations[pkgKey] = "true";
-
-    // Apply the updated Namespace
-    log.debug(`Updating namespace ${pkg.metadata.namespace} with istio injection label`);
-    await K8s(kind.Namespace).Apply(
-      {
-        metadata: {
-          name: pkg.metadata.namespace,
-          labels,
-          annotations,
-        },
-      },
-      { force: true },
+    log.info(
+      `Updating package-specific annotation for ${pkg.metadata.name} in namespace ${pkg.metadata.namespace}.`,
     );
+  }
 
-    // Kill the pods if we changed the value of the istio-injection label
-    if (originalInjectionLabel !== labels[injectionLabel]) {
-      log.debug(
-        `Attempting pod restart in ${pkg.metadata.namespace} based on istio injection label change`,
-      );
-      await killPods(pkg.metadata.namespace, true);
-    }
+  // Apply namespace updates if there are changes
+  const updatedNamespace = {
+    metadata: {
+      name: pkg.metadata.namespace,
+      labels,
+      annotations,
+    },
+  };
+
+  if (
+    JSON.stringify(sourceNS.metadata?.labels) !== JSON.stringify(labels) ||
+    JSON.stringify(sourceNS.metadata?.annotations) !== JSON.stringify(annotations)
+  ) {
+    log.debug(`Applying updates to namespace ${pkg.metadata.namespace}.`);
+    await K8s(kind.Namespace).Apply(updatedNamespace, { force: true });
+  } else {
+    log.debug(`No namespace updates needed for ${pkg.metadata.namespace}.`);
+  }
+
+  // Restart pods if required
+  if (shouldRestartPods) {
+    log.debug(`Restarting pods in ${pkg.metadata.namespace} due to configuration changes.`);
+    await killPods(pkg.metadata.namespace, true);
   }
 }
 
```

Comment:
I can't comment on the actual line, but right below here we have a `cleanupNamespace` that gets run on removal of a Package CR. We should make sure we delete our `nativeIstioAnnotation` in that function, around the same place we delete the injectionAnnotation (`delete annotations[injectionAnnotation];`).